sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py

```diff
@@ -247,6 +247,8 @@ class RequestsTransport(HttpTransport):
     def __init__(self, **kwargs) -> None:
         self.session = kwargs.get("session", None)
         self._session_owner = kwargs.get("session_owner", True)
+        if not self._session_owner and not self.session:
```

Comment:
Fix for https://github.com/Azure/azure-sdk-for-python/issues/31945

Found by pyright

Reply:
Shouldn't we add this error to the other transports, like aiohttp?
While it may not be a typing error elsewhere, I'm pretty sure they also have the same omission - from what I can see aiohttp will have a similarly weird error during open/send.

Reply:
Yep, good point, pushing an update